src/Logging/src/DynamicLogger/DynamicConsoleLoggerProvider.cs

```diff
@@ -23,66 +23,43 @@
+        private readonly ConcurrentDictionary<string, LogLevel> _originalLevels = new ConcurrentDictionary<string, LogLevel>();
+        private readonly ConcurrentDictionary<string, Func<string, LogLevel, bool>> _runningFilters = new ConcurrentDictionary<string, Func<string, LogLevel, bool>>();
 
+        private readonly IOptionsMonitor<LoggerFilterOptions> _filterOptions;
+        private readonly IEnumerable<IDynamicMessageProcessor> _messageProcessors;
+        private Func<string, LogLevel, bool> _filter = _falseFilter;
         private ConcurrentDictionary<string, DynamicConsoleLogger> _loggers = new ConcurrentDictionary<string, DynamicConsoleLogger>();
         private ConsoleLoggerProvider _delegate;
-        private IConsoleLoggerSettings _settings;
-        private IOptionsMonitor<LoggerFilterOptions> _filterOptions;
-        private IEnumerable<IDynamicMessageProcessor> _messageProcessors;
 
         private bool disposed = false;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="DynamicLoggerProvider"/> class.
-        /// </summary>
-        /// <param name="settings">Logging Settings</param>
-        public DynamicLoggerProvider(IConsoleLoggerSettings settings)
-        {
-            _delegate = new ConsoleLoggerProvider(settings);
-            _settings = settings;
-            SetFiltersFromSettings();
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="DynamicLoggerProvider"/> class.
-        /// </summary>
-        /// <param name="filter">Default log level filter</param>
-        /// <param name="includeScopes">Enable log scoping</param>
-        public DynamicLoggerProvider(Func<string, LogLevel, bool> filter, bool includeScopes)
-        {
-            _delegate = new ConsoleLoggerProvider(filter, includeScopes);
-            _filter = filter ?? _falseFilter;
-            _settings = null;
-        }
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="DynamicLoggerProvider"/> class.
+        /// Initializes a new instance of the <see cref="DynamicConsoleLoggerProvider"/> class.
         /// </summary>
         /// <param name="options">Pass-through to ConsoleLoggerProvider constructor</param>
         /// <param name="filterOptions">Logger filters</param>
         /// <param name="messageProcessors">message processors to apply to message</param>
-        public DynamicLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IOptionsMonitor<LoggerFilterOptions> filterOptions, IEnumerable<IDynamicMessageProcessor> messageProcessors = null)
+        public DynamicConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options, IOptionsMonitor<LoggerFilterOptions> filterOptions, IEnumerable<IDynamicMessageProcessor> messageProcessors = null)
         {
             _filterOptions = filterOptions;
-            SetFiltersFromOptions();
             _delegate = new ConsoleLoggerProvider(options);
```

Comment:
They could, and yes we'd probably want to avoid that. Is there a straightforward way to avoid that situation?

Reply:
It might make sense to abandon the idea of wrapping the logger altogether and *just* register a filter. https://github.com/aspnet/Extensions/blob/d656c4f7e22d1c0b84cab1b453c50ce73c89a071/src/Logging/Logging/src/FilterLoggingBuilderExtensions.cs#L20

You could keep something like this class around as a DI service to manage configuration, and then just register a filter around the console logging provider to do your work. Then you can always avoid duplication.

Reply:
@rynowak I like that idea, and I was hoping for something simpler than what we've got, but there's a detail that I tend to forget about: in the Dynamic Console Logger, we also provide a hook for providing additional message processing before passing the log entry to the underlying logger. The main (only?) use case we have right now is for enabling log correlation with OpenCensus -- https://github.com/SteeltoeOSS/steeltoe/blob/master/src/Management/src/TracingBase/TracingLogProcessor.cs we'd need an alternative mechanism for attaching metadata for basically all log entries to move in that direction